src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx

```diff
@@ -188,6 +190,24 @@ export function DashboardApp({
     getScreenshotContext,
   ]);
 
+  useMemo(() => {
+    if (!dashboardAPI) return;
+    dashboardAPI?.expandedPanelId.subscribe(() => {
+      const expandedPanel = dashboardAPI.expandedPanelId.value;
+      if (expandedPanel) {
+        return history.replace({
+          pathname: `${createDashboardEditUrl(dashboardAPI.savedObjectId.value)}/${expandedPanel}`,
+          search: history.location.search,
```

Comment:
`history.location` has more than just `search` - to see this, add `console.log(history.location)` in this subscribe. I think we should spread `history.location`, instead, so that we don't lose the rest of the location.

Reply:
I'm not sure how we would do this... I'm looking at the LocationDescriptiorObject in History:
```
export interface LocationDescriptorObject<S = LocationState> {
    pathname?: Pathname | undefined;
    search?: Search | undefined;
    state?: S | undefined;
    hash?: Hash | undefined;
    key?: LocationKey | undefined;
}
```

Without adding the search with history.location.search, the new url does not include the search state and doing the console log on history.location is showing me the pathname and the search properties but state isn't defined. 


console.log when opening a dashboard without the expanded panel
![Screenshot 2024-09-05 at 12 55 31 PM](https://github.com/user-attachments/assets/6185763a-ca80-4e88-aded-07499745d94b)

console.log when opening the expanded panel 
![Screenshot 2024-09-05 at 12 53 15 PM](https://github.com/user-attachments/assets/7507bad3-b846-4e22-aea9-6421cfbc7989)

Reply:
Sorry yeah, you should still overwrite `pathname` - but I think we should keep the **rest** of `history.location` intact (not just `history.search`). Does that make sense?

Reply:
Thank you for zooming with me and explaining - for my own learning, spreading the history.location before setting any of the history.replace() options to be overridden 👍🏻